docs/HANDLED_RULES.md

```diff
@@ -259,6 +260,44 @@ class SynchronizationOnGetClass {
 ```
 
 -----
+
+#### Variables should not be self-assigned ([Sonar Rule 1656](https://rules.sonarsource.com/java/type/Bug/RSPEC-1656))
+
+Any assignment with identical left and right expressions will be processed. If the identifier being using in the self-assignment exists as both a local variable and a field, then the left expression will be changed by adding this. at the beginning of the expression. In any other case, including cases where there are invocations or access to another class field, such as objectA.b = objectA.b, the assignment will be removed.
```

Comment:
"being using" -> "being used"
"this." -> "`this.`"
"objectA.b = objectA.b" -> "`objectA.b = objectA.b`"